### PR TITLE
Add gcloud authentication/configuration

### PR DESCRIPTION
**Pull request by benmoss**
Thought these were just magic env vars, apparently not!

/assign @neolit123

### CODE REVIEW

File: kubeadm/hack/ci-e2e.sh [hunk 1]
```diff
@@ -67,6 +67,32 @@ mkdir -p "$ARTIFACTS/logs/"
...
 HEART_BEAT_PID=$(echo $!)
 
+if [[ -z "$GOOGLE_APPLICATION_CREDENTIALS" ]]; then
+	cat <<EOF
+$GOOGLE_APPLICATION_CREDENTIALS is not set.
+Please set this to the path of the service account used to run this script.
+EOF
+	exit 2
+else
+	gcloud auth activate-service-account --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
```
Comment (neolit123) [inline kubeadm/hack/ci-e2e.sh]: isn't the similar logic from runner.sh triggering?
https://github.com/kubernetes/test-infra/blob/master/images/bootstrap/runner.sh

i do see this in the recent logs:
> Activated service account credentials for: [pr-kubekins@kubernetes-jenkins-pull.iam.gserviceaccount.com]

File: kubeadm/hack/ci-e2e.sh [hunk 2]
```diff
@@ -67,6 +67,32 @@ mkdir -p "$ARTIFACTS/logs/"
...
+GCP_PROJECT is not set. Using project_id $GCP_PROJECT
+EOF
+fi
+if [[ -z "$GCP_REGION" ]]; then
+	cat <<EOF
+GCP_REGION is not set.
+Please specify which the GCP region to use.
+EOF
+	exit 2
+fi
```
Comment (neolit123) [inline kubeadm/hack/ci-e2e.sh]: i cannot seem to find where are we (or what is) setting GCP_REGION.

Comment (benmoss) [reply kubeadm/hack/ci-e2e.sh]: Added a default

File: kubeadm/hack/ci-e2e.sh (hunk 1 above)
Comment (benmoss) [reply kubeadm/hack/ci-e2e.sh]: Removed this since yeah I think it's unnecessary.

**Pull request merged**